controllers/datafoundationclient.go

```diff
@@ -47,16 +50,29 @@ func dfcSetupWatches(controllerBuilder *builder.Builder) {
 		),
 	)
 	controllerBuilder.
-		Owns(&opv1a1.ClusterServiceVersion{}, csvPredicates).
-		Owns(&ocsclient.StorageClient{}).
-		Owns(&ocsclient.StorageClassClaim{})
+		Owns(&opv1a1.ClusterServiceVersion{}, csvPredicates)
+
+	if utils.MapCRDAvailability(storageClientCRD)[storageClientCRD] {
+		doesCRDExistsAtStart = true
+		controllerBuilder.
+			Owns(&ocsclient.StorageClient{}).
+			Owns(&ocsclient.StorageClassClaim{})
+	}
 }
 
 func DFCAddToScheme(scheme *runtime.Scheme) {
 	utilruntime.Must(ocsclient.AddToScheme(scheme))
 }
 
 func dfcReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
+	// Check if the required CRDs are installed or not
+	if !doesCRDExistsAtStart {
```

Comment:
Why do we care about existence? what we do care about is if we are watching or not watching specific CRs and we already know that this is based on the CRD map that was loaded in `main.go` and provided to the reconciler.